js/style/style.js

```diff
@@ -370,7 +370,10 @@ Style.prototype = util.inherit(Evented, {
         layer = this.getReferentLayer(layer);
         layer.setLayoutProperty(name, value);
         this._broadcastLayers();
-        this.sources[layer.source].reload();
+        if (layer.source) {
+            this.sources[layer.source].reload();
+        }
+        this.fire('change');
```

Comment:
Let's not do this in this PR. I am not sure it makes sense, and if it does make sense it should be done for the other mutating methods such as `setPaintProperty`.